### ISSUE TITLE
Feature Request: access gradients for a specific layer

### ISSUE THREAD

**Issue by xksteven** [tensorflow/tensorflow#7800]
Currently tensorflow will provide the gradients for the entire network, i.e. getting grads_and_vars will return all of the gradients but there isn't any nice way to specifically get the gradients of a specific layer.

An example use case:  For the paper Neural Networks are easily fooled, they took the gradients from the last layer and added them to the original image.

There is no current way to do this in tensorflow as far as I know.

**Comment by yaroslavvb** [comment]
You can get gradients for any layer, `tf.gradients(loss, layer2)` will give you gradients at layer2 . (ps this q is better suited for stackoverflow)

**Comment by xksteven** [comment]
Sorry I didn't know it was already implemented.  I'll read the documentation more thoroughly next time.
